pkg/registry/pod/strategy.go

```diff
@@ -31,11 +31,39 @@ import (
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/registry/generic"
+	nodeetcd "k8s.io/kubernetes/pkg/registry/node/etcd"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util"
 	"k8s.io/kubernetes/pkg/util/fielderrors"
 )
 
+type HostLocator interface {
```

Comment:
This interface helps a lot, thanks-- but I still kind of feel like the implementation should be in the node package, and it should be injected here, so that pod & pod/etcd packages do not need to import the node or node/etcd packages. Does that make sense? Is it feasible?